### ISSUE TITLE
Norm operator with string input on a one dimensional array results in cryptic error

### ISSUE THREAD

**Issue by KerimovEmil** [cvxpy/cvxpy#1865]
**Describe the bug**
Norm operator with string input on a one dimensional array results in cryptic error. Specifically the norm atom doesn't correctly handle the cases of when `x` is a one dimensional vector with the options of `fro` or `nuc`. It passes `p` directly into the `pnorm` atom which tries to then compare the string against the number 0, resulting in a type error. 

If it makes sense I can create a PR to handle those two cases, if `x` is not a matrix. For the frobenius norm at least I think it should default to the 2-norm if `x` is a vector. 

**To Reproduce**

```
import cvxpy
w = cvxpy.Variable(10)
t = cxvpy.norm(w, p='fro')
```

**Expected behavior**
I expected either an error message specifying that the frobenius option is not available for vector inputs or even better for it to default to the 2 norm after casting `x` as a vector. 

**Output**
![image](https://user-images.githubusercontent.com/8174342/184509338-c910f0d0-688b-4c42-867a-1700e1373134.png)

**Version**
 - OS: windows
 - CVXPY Version: 1.2

**Additional context**
This issue was found when I tried upgrading some existing code from the older 0.4 cvxpy version to the latest 1.2. In the older code since it didn't check for the dimension of `x`, the code resulted in just defaulting to `pnorm(vec(x), 2)`.

**Comment by rileyjmurray** [comment]
Hi @KerimovEmil, thanks for the report. I'm not sure what our semantics should be here. The ``pnorm`` function is supposed to be restricted to vectors or applied along an axis to a matrix. Based on that, strings specifying matrix norms (like the Frobenius or nuclear norms) wouldn't be acceptable arguments. @SteveDiamond @phschiele @akshayka do you suppose we should raise an error here?

**Comment by phschiele** [comment]
@rileyjmurray Raising an informative error message seems like a good idea.

**Comment by SteveDiamond** [comment]
To be clear this isn't calling ``pnorm`` externally. It's about the behavior of ``norm``. I would vote for @KerimovEmil's idea of ``fro`` mapping to the L2 norm and ``nuc`` raising an error. It's pretty clear what the user meant for ``fro`` norm even if it's not strictly correct semantics.

**Comment by SteveDiamond** [comment]
Resolved by #1871
